### DISCUSSION TITLE
Problem while creating pdf: It seems webp-files are generated instead of jpg - could that be the reason no pdf can be created?

### DISCUSSION THREAD

**Discussion by narzalus** [Lofter1/anyflip-downloader#47] [General]
Hi, I tried this downloader (amazing work btw, thank you!) recently and wasn't able to create a proper pdf with some files - there's just a pdf-file with a size of 1kb.
The temp folder is created, but instead of jpg-files there are webp-files downloaded (unfortunately the temp folder is deleted when trying to create the pdf). 
So could it be is it possible that (with newer uploads maybe?) webp-files are generated in the temp folder instead of jpg? And therefore it can't convert properly to pdf?
Would there be a solution for the downloader to convert those webp-files into pdf-files? Or to at least be able to keep the temp folder to convert those files manually into a pdf?

Thanks in advance!

**Comment by MySephiroth91** [comment]
I second this request: back in december I used the downloader and it all went fine, but now the only result is a 1 kbt-sized blank pdf, and nothing else.

**Comment by Lofter1** [comment]
Hey, sorry for not responding to anything on here for a while. Life's been really stressful and it doesn't seem to slow down. 

Skipping the deletion step using a flag should be relatively easy from a coding perspective. I will look into implementing this asap. I'm not using this downloader much myself anymore, so it's become a low priority for me, so anything beside that might take some time. I will try to merge all the pull requests and then try to implement this flag. And then, depending on my available time, see if I can generate a PDF automatically using formats that aren't jpeg.

**Reply by narzalus** [reply]
Hey, no worries, just wanted to point it out - it's not that it's super important! :  ) I already thought so anyway that it's (understandibly) not one of your priorites. Thank you for your reply though!

**Comment by MySephiroth91** [comment]
I'm sorry to insist, but apparently nothing has changed: it still keeps downloading a blank pdf, and the download folder still deletes itself. Am I missing something ?

**Reply by Lofter1** [reply]
Have you used the flag `keep-download-folder`? An example for this would be `anyflip-downloader -keep-download-folder https://anyflip.com/my/book`. With that flag, the folder shouldn't be deleted anymore (keep in mind that you need to update anyflip-downloader first!). I have not updated the program to be able to handle the different file format for now (I'm not sure if the PDF converting library supports anything but jpeg). Sadly, my life just got even more stressful just a few hours ago so I will not have the time to implement this feature, but one might be able to find a program that can convert the webp files in the download folder into a PDF. I will also gladly look into merging any PR that addresses this issue.

**Reply by MySephiroth91** [reply]
Oh, that was what I was missing ! Thank you !

**Comment by DSO-13** [comment]
Has this been fixed at some point? Because yesterday I was having issues with the 1kb compiled PDFs, then I found this discussion thread, ran the download/installer powershell command, and tried again with the keep-download-folder flag just expecting to preserve the webp pics so I could try to compile them myself, and all of a sudden it works entirely correctly with a proper and complete outputted pdf from this downloader

Which means 1 of 2 things

1) updates subsequent to this discussion have corrected the issue and either I'm blind and didn't see that change listed or it wasn't noted anywhere else

or

2) for weird black magic reasons I can't explain the keep-download-folder flag somehow overcomes the downloader's problem with compiling webp images into a pdf. This makes less sense than option 1 but would also not be the weirdest thing I've ever seen.
